source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,253 @@
+
+You can use a **cursor** to iterate over the documents returned by the ``Find()``
+method. A cursor is a mechanism that allows an application to iterate over database 
+results while holding only a subset of them in memory at a given time. Cursors
+are useful when your ``Find()`` method returns a large amount of documents.
+
+To use a cursor to iterate over the documents, pass a 
+query filter to the ``Find()`` method that specifies the criteria of the documents 
+you want to find, then chain the ``ToCursor()`` or ``ToCursorAsync()`` method as follows:
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: find-cursor-async
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var restaurants = await _restaurantsCollection.Find(filter).ToCursorAsync();
+
+   .. tab:: Synchronous
+      :tabid: find-cursor-sync
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var restaurants = _restaurantsCollection.Find(filter).ToCursor();
+
+If you are returning a small number of documents, or need your results returned
+as a ``List`` object, use the ``ToList()`` or ``ToListAsync()`` methods.
+
+To find multiple documents in a collection and hold them in memory as a list, pass a query filter 
+to the ``Find()`` method that specifies the criteria of the documents you want 
+to find, then chain the ``ToList()`` or ``ToListAsync()`` method as follows:
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: find-list-async
+
+      .. code-block:: csharp
+
+         var restaurants = await _restaurantsCollection.Find(filter).ToList();
+
+   .. tab:: Synchronous
+      :tabid: find-list-sync
+
+      .. code-block:: csharp
+
+         var restaurants = _restaurantsCollection.Find(filter).ToListAsync();
+
+To see a full example of using the ``Find()`` method to find multiple documents,
+see :ref:`csharp-retrieve-additional-information`.
+
+Find All Documents
+~~~~~~~~~~~~~~~~~~
+
+To find all documents in a collection, pass an empty
+``BsonDocument`` as a query filter. Then chain the ``ToList()`` or ``ToListAsync()``
+method to return the documents as a ``List`` object, or the ``ToCursor()`` or ``ToCursorAsync()``
+method to return the documents as a cursor.
+
+To see a fully runnable example of using the ``Find()`` method to find all documents, see 
+:ref:`csharp-retrieve-additional-information`.
+
+Modify Find Behavior
+~~~~~~~~~~~~~~~~~~~~
+
+You can modify the behavior of the ``Find()`` method by passing
+a ``FindOptions`` object.
+
+You can configure the commonly used options with the following methods:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``BatchSize`` 
+     - | Gets or sets the number of documents to return per batch.
+
+   * - ``Collation`` 
+     - | Sets the collation options. 
+
+   * - ``Comment`` 
+     - | Sets the comment to the query.
+
+   * - ``Hint`` 
+     - | Sets the hint for which index to use.
```

Comment:
Suggestion: These definitions are largely circular. I believe all four of the definitions highlighted define the option using the option name (not super helpful if you don't already know what the option name is referring to).

Consider providing an explanation of each option that helps give a reader something to relate the option to, or providing a link to let the reader learn more about what the option is doing:

```suggestion
     - | Gets or sets the number of documents hold in a cursor at a given time.

   * - ``Collation`` 
     - | Sets the collation options. To learn more about collations, see the TODO: collations page. 

   * - ``Comment`` 
     - | Sets the comment to the query. To learn more about query comments, see https://www.mongodb.com/docs/manual/reference/operator/query/comment/.

   * - ``Hint`` 
     - | Sets the hint for which index to use.
```

Reply:
Added all but the collation link since we don't have a plan of making a collations page in the MVP. Happy to add it when we do add that page